src/util/OAuthErrors.ts

```diff
@@ -23,15 +23,15 @@ class TypedOAuthError<T extends ErrorType> extends OAuthError {
   orginalError: AuthSdkError | SdkOAuthError;
   errorDetails: ErrorDetails
 
-  constructor(originalError, errorTypeCtor: new () => T) {
+  constructor(originalError: AuthSdkError | SdkOAuthError, errorTypeCtor: new () => T) {
     super(originalError.message);
     this.errorType = new errorTypeCtor();
     this.orginalError = originalError;
 
     this.errorDetails = {
       errorSummary: this.getErrorSummary(),
       errorCode: originalError.errorCode,
-      errorCauses: originalError.errorCauses,
+      errorCauses: 'errorCauses' in originalError ? originalError.errorCauses : undefined,
```

Comment:
Does the default type of `originalError.errorCauses` not work here?

Reply:
The reason TS complains about this is because the type of the original error can be either `AuthSdkError` or `SdkOAuthError`. `AuthSdkError` contains errorCauses whereas `SdkOAuthError` does not, so I had to perform the check to ensure `errorCauses` exists before trying to use it. I believe this complains because strict type checking is enabled in v3, so when we use this util from v2 it enforces the strict type checking.

Reply:
Interesting, since we sometimes return `undefined` in the ternary and TS is ok with that, I'd have expected it to be OK with `originalError.errorCauses` on its own since that would return `errorCauses` or `undefined`

Reply:
Yea, I thought the exact same thing, I was confused why it was even complaining in the first place, I made this update on a whim and it seemed to appease the TS gods 😅 

Removing the logic: 
![image](https://user-images.githubusercontent.com/97472729/230423682-962b3890-6f16-4c05-bc3e-f942b85e28af.png)

Re-adding the logic:
![image](https://user-images.githubusercontent.com/97472729/230423883-e3b9b5b2-0997-4bee-868f-6785cb130b7d.png)